index.js

```diff
@@ -30,9 +13,18 @@ module.exports.val = function (count, fn) {
 		current = fn();
 
 		if (current !== first) {
-			return current;
+			return val ? current : false;
 		}
+
 	}
 
-	return first;
+	return val ? first : true;
+}
+
+module.exports = function (count, fn) {
```

Comment:
That's really elegant--however, I cannot get tests to pass that way. Is there an obvious mistake I am making?

```javascript
'use strict';

function stable(val, count, fn) {
	if (fn === undefined) {
		fn = count;
		count = 1000;
	}

	var current;
	var first = fn();

	for (var i = 1; i < count; i++) {
		current = fn();

		if (current !== first) {
			return val ? current : false;
		}

	}

	return val ? first : true;
}

module.exports = stable.bind(false);
module.exports.val = stable.bind(true);
```

Errors "undefined is not a function"